build-tools/packages/build-cli/src/handlers/checkFunctions.ts

```diff
@@ -477,6 +478,91 @@ export const checkAssertTagging: StateHandlerFunction = async (
 	return true;
 };
 
+/**
+ * Checks that release notes have been generated.
```

Comment:
This is not what this function actually does (ex: for example when not a server or client release). I think it does:

```suggestion
 * Logs an error and returns false iff release notes are required and not already generated.
```

Reply:
Maybe even document that its a no-op in test mode? That choice seems odd since it doesn't let you test this, but as noted below, I'm not very clear on test-mode's semantics

Reply:
Added more clarification about the behavior of the function and how it interacts with the state machine.